db/migrate/20150213162341_add_value_constraints_batch_1.rb

```diff
@@ -0,0 +1,1311 @@
+
+-- decide on level and units constraints
+
+ALTER TABLE managements ALTER COLUMN notes SET NOT NULL,
+                        ALTER COLUMN notes SET DEFAULT '';
+
+
+/* METHODS */
+
+ALTER TABLE methods ALTER COLUMN name SET NOT NULL;
+ALTER TABLE methods ADD CONSTRAINT normalized_method_name CHECK (is_whitespace_normalized(name));
+ALTER TABLE methods ALTER COLUMN description SET NOT NULL;
+-- Additional constraints will be added when candidate key is implemented.
+
+
+/* MIMETYPES */
+
+-- Constraints are in AddUniquenessConstraints migration.
+
+
+/* MODELS */
+
+ALTER TABLE models ALTER COLUMN model_name SET NOT NULL,
+                   ADD CONSTRAINT no_spaces_in_model_name CHECK (is_whitespace_free(model_name));
+ALTER TABLE models ALTER COLUMN revision SET NOT NULL,
+                   ADD CONSTRAINT normalized_revision_specifier CHECK (is_whitespace_normalized(revision));
+
+
+/* MODELTYPES */
+
+ALTER TABLE modeltypes ADD CONSTRAINT no_spaces_in_modeltype_name CHECK (is_whitespace_free(name));
+
+
+/* MODELTYPES_FORMATS */
+
+ALTER TABLE modeltypes_formats ADD CONSTRAINT valid_modeltype_format_tag CHECK (tag ~ '^[a-z]+$');
+ALTER TABLE modeltypes_formats ALTER COLUMN required SET NOT NULL;
+ALTER TABLE modeltypes_formats ALTER COLUMN input SET NOT NULL;
+
+
+/* PFTS */
+
+ALTER TABLE pfts ALTER COLUMN definition SET NOT NULL;
+ALTER TABLE pfts ADD CONSTRAINT normalized_pft_name CHECK (is_whitespace_normalized(name));
+ALTER TABLE pfts ALTER COLUMN pft_type SET NOT NULL,
+                 ADD CONSTRAINT valid_pft_type CHECK (pft_type IN ('plant', 'cultivar', ''));
+
+/* PRIORS */
+
+ALTER TABLE priors ALTER COLUMN variable_id SET NOT NULL;
+ALTER TABLE priors ALTER COLUMN phylogeny SET NOT NULL,
+                   ADD CONSTRAINT normalized_prior_phylogeny_specifier CHECK (is_whitespace_normalized(phylogeny));
+ALTER TABLE priors ALTER COLUMN distn SET NOT NULL,
+                   ADD CONSTRAINT valid_prior_distn CHECK (distn IN ('beta', 'binom', 'cauchy', 'chisq', 'exp', 'f', 'gamma', 'geom', 'hyper', 'lnorm', 'logis', 'nbinom', 'norm', 'pois', 't', 'unif', 'weibull', 'wilcox'));
+ALTER TABLE priors ALTER COLUMN parama SET NOT NULL;
+ALTER TABLE priors ADD CONSTRAINT nonnegative_prior_sample_size CHECK (n >= 0);
+
+/* PROJECTS */
+
+ALTER TABLE projects ALTER COLUMN name SET NOT NULL,
+                     ADD CONSTRAINT normalized_project_name CHECK (is_whitespace_normalized(name));
+ALTER TABLE projects ALTER COLUMN outdir SET NOT NULL;
+ALTER TABLE projects ALTER COLUMN description SET NOT NULL;
+
+
+/* RUNS */
+
+ALTER TABLE runs ALTER COLUMN outdir SET NOT NULL,
+                 ALTER COLUMN outdir SET DEFAULT '';
+ALTER TABLE runs ALTER COLUMN outprefix SET NOT NULL,
+                 ALTER COLUMN outprefix SET DEFAULT '';
+ALTER TABLE runs ALTER COLUMN setting SET NOT NULL,
+                 ALTER COLUMN setting SET DEFAULT '';
+ALTER TABLE runs ALTER COLUMN started_at SET NOT NULL;
```

Comment:
the started_at and finished_at can be null, when the run is created these are not set. These are set when the run is actually started and finished.